### DISCUSSION TITLE
ArrayCards 根据所在的位置 和 总数，来动态设置className的方式

### DISCUSSION THREAD

**Discussion by cl1069573062** [alibaba/formily#2744] [Q&A 2.x] [answered]
![image](https://user-images.githubusercontent.com/15069666/148682179-4f1f00d9-c595-4ca9-9dda-b5e3b89f5070.png)

我希望根据arraycards，索引和总数来动态设置className,
![image](https://user-images.githubusercontent.com/15069666/148682206-fcb35610-d2ef-47b5-b921-7204ccea959a.png)
目前的做法是，修改源码
![image](https://user-images.githubusercontent.com/15069666/148682217-ea1b1ed1-cc1b-46f8-8b1e-cd3bedb58e7d.png)
是否有更好的不侵入的方式

**Comment by janryWang** [comment, marked as answer]
"x-component-props":{className:"{{$index === 0 ? 'xxx' : 'yyy'}}"}

**Reply by chenWebCoder** [reply]
请问想要根据 index 动态设置 x-disabled 属性，需要在内部如何获取到 index 呢？

Demo 地址：https://codesandbox.io/s/nifty-mccarthy-0kkzfq?file=/App.tsx

**Reply by chenWebCoder** [reply]
补充下：这里的 "x-component-props":{className:"{{$index === 0 ? 'xxx' : 'yyy'}}"} 这种方式任然适用于设置 disabled，但是想使用类似 ArrayCards?.useIndex?.() 这种方式获取 index;
